src/registrar/management/commands/send_domain_invitations.py

```diff
@@ -0,0 +1,148 @@
+        )
+
+    def handle(self, **options):
+        """Process the objects in TransitionDomain."""
+
+        logger.debug("checking domains and preparing emails")
+        # Get all TransitionDomain objects
+        self.transition_domains = TransitionDomain.objects.filter(
+            email_sent=False,
+        ).order_by("username")
+
+        self.build_emails_to_send_array()
+
+        if options["send_emails"]:
+            logger.debug("about to send emails")
+            self.send_emails()
+            logger.debug("done sending emails")
+
+            self.update_domains_as_sent()
+
+            logger.debug("done sending emails and updating transition_domains")
+        else:
+            logger.debug("not sending emails")
+
+    def build_emails_to_send_array(self):
+        """this method sends emails to distinct usernames"""
+
+        # data structure to hold email context for a single email;
+        # transition_domains ordered by username, a single email_context
+        # may include information from more than one transition_domain
+        email_context = {"email": ""}
+
+        # loop through all transition_domains; group them by username
+        # into emails_to_send_array
+        for transition_domain in self.transition_domains:
+            # attempt to get the domain from domain objects; if there is
+            # an error getting the domain, skip this domain and add it to
+            # domains_with_errors
+            try:
+                domain = Domain.objects.get(name=transition_domain.domain_name)
+                # if prior username does not match current username
+                if (
+                    not email_context["email"]
+                    or email_context["email"] != transition_domain.username
+                ):
+                    # if not first in list of transition_domains
+                    if email_context["email"]:
+                        # append the email context to the emails_to_send array
+                        self.emails_to_send.append(copy.deepcopy(email_context))
+                    email_context["domains"] = []
+                email_context["email"] = transition_domain.username
+                email_context["domains"].append(
+                    {
+                        "name": transition_domain.domain_name,
+                        "url": settings.BASE_URL
+                        + reverse("domain", kwargs={"pk": domain.id}),
+                    }
+                )
+            except Exception as err:
+                # error condition if domain not in database
+                self.domains_with_errors.append(
+                    copy.deepcopy(transition_domain.domain_name)
+                )
+                logger.error(
+                    f"error retrieving domain {transition_domain.domain_name}: {err}"
+                )
+        # if there are at least one more transition domains than errors,
+        # then send one more
```

Comment:
```suggestion
        # then append one more item
```
This seemed a bit unclear to me, I would recommend something like above